### DISCUSSION TITLE
ETF Holding Informations

### DISCUSSION THREAD

**Discussion by dustinspace** [ranaroussi/yfinance#1761] [Ideas]
Hi everybody,
It would like great being able to download the list of the "top 10 holdings" for the ETF Founds (https://finance.yahoo.com/quote/EUNL.DE/holdings?p=EUNL.DE): "Symbol" and "% Asset" data. 

What do you think ? 
Anyway if somebody could help me to find the API link, I could try to write a test code for this library ....

Thanks, Dustin.

**Comment by ValueRaider** [comment]
> help me to find the API link

Firefox -> URL -> F12 -> Network tab -> F5

[a comment by an automated account is not shown]

**Comment by ericpien** [comment]
@dustinspace, following https://github.com/ranaroussi/yfinance/pull/2041, you are now able to query etf top holdings via:
```
eunl = yf.Ticker('EUNL.DE')
data = eunl.funds_data
data.top_holdings
```

<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Name</th>
      <th>Holding Percent</th>
    </tr>
    <tr>
      <th>Symbol</th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>AAPL</th>
      <td>Apple Inc</td>
      <td>0.048192</td>
    </tr>
    <tr>
      <th>MSFT</th>
      <td>Microsoft Corp</td>
      <td>0.043683</td>
    </tr>
    <tr>
      <th>NVDA</th>
      <td>NVIDIA Corp</td>
      <td>0.043275</td>
    </tr>
    <tr>
      <th>AMZN</th>
      <td>Amazon.com Inc</td>
      <td>0.025895</td>
    </tr>
    <tr>
      <th>META</th>
      <td>Meta Platforms Inc Class A</td>
      <td>0.015390</td>
    </tr>
    <tr>
      <th>GOOGL</th>
      <td>Alphabet Inc Class A</td>
      <td>0.014953</td>
    </tr>
    <tr>
      <th>GOOG</th>
      <td>Alphabet Inc Class C</td>
      <td>0.013073</td>
    </tr>
    <tr>
      <th>AVGO</th>
      <td>Broadcom Inc</td>
      <td>0.010484</td>
    </tr>
    <tr>
      <th>TSLA</th>
      <td>Tesla Inc</td>
      <td>0.009840</td>
    </tr>
    <tr>
      <th>LLY</th>
      <td>Eli Lilly and Co</td>
      <td>0.009614</td>
    </tr>
  </tbody>
</table>

**Comment by ohjho** [comment]
have you looked into the [etfpy library](https://github.com/JakubPluta/pyetf)?
